Add Elixir.tmbundle

jish opened the pull request:
I have never created a Homebrew Cask before so I'm not sure if this is
correct or not. I used some examples in this repo and I also used the
"Adding a Cask" documentation.

Please let me know what I should change or how I should test this to
make sure it works properly.

jish committed:
Add Elixir.tmbundle

I have never created a Homebrew Cask before so I'm not sure if this is
correct or not. I used some examples in this repo and I also used the
"Adding a Cask" documentation.

Please let me know what I should change or how I should test this to
make sure it works properly.

jish committed:
Turn off SHA checking for `tm-elixir`.

https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#special-value-no_check

> "The special value `sha256 :no_check` is used to turn off SHA
> checking whenever checksumming is impractical due to the upstream
> configuration."

> "`version :latest` requires `sha256 :no_check`, and this pairing is
> common."

claui commented on Casks/tm-elixir.rb:
```diff
@@ -0,0 +1,11 @@
+  name "A TextMate Bundle for the Elixir programming language."
+  homepage "https://github.com/elixir-lang/elixir-tmbundle"
+
+  tmbundle "Elixir.tmbundle"
```
This line causes the following error:

> ==> Installing Cask tm-elixir
> Error: It seems the TextMate bundle source '/usr/local/Caskroom/tm-elixir/latest/Elixir.tmbundle' is not there.
> Error: Install incomplete.

The reason for this is that extracting `master.tar.gz` results in a directory named `elixir-tmbundle-master` while Homebrew looks for something called `Elixir.tmbundle`, which is obviously not there.

To fix this, you can just edit this line so it says:

    tmbundle "elixir-tmbundle-master", target: "Elixir.tmbundle"

That way, Homebrew will find the source directory by the odd name. Also, the `:target` option will instruct it to rename the directory to `Elixir.tmbundle` proper inside TextMate’s `Bundles` directory.

claui commented:
Great work @jish, and even more so for a first Cask!

The best option for Casks is to always pin to a version, or release; this allows users to benefit from (and ultimately trust) the SHA sums. If we’re dealing with a Cask that must use `version :latest` (e. g. when the upstream project does not define releases and version numbers), then the Cask needs`sha256 :no_check`, which you did correctly. 👍

As soon as the `It seems the TextMate bundle source 'Elixir.tmbundle' is not there` error is fixed, there’s nothing stopping us from merging!

jish committed:
Reference unarchived Elixir folder properly

The Elixir archive extracts into a folder called
`elixir-tmbundle-master`. We then want to use that folder and place it
at `Elixir.tmbundle` so TextMate can interpret it properly.

jish commented:
Thanks for the help, I updated the `tmbundle` line.

claui commented:
Merged.
